(base) PS C:\Users\hperu> conda activate flwr_tf
(flwr_tf) PS C:\Users\hperu>  cd .\OneDrive\Desktop\fl\FLOWER\6C_20rounds\
(flwr_tf) PS C:\Users\hperu\OneDrive\Desktop\fl\FLOWER\6C_20rounds> python .\server.py
2025-05-29 18:05:04.398594: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-05-29 18:05:05.361721: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
INFO flwr 2025-05-29 18:05:08,212 | app.py:148 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2025-05-29 18:05:08,231 | app.py:168 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2025-05-29 18:05:08,232 | server.py:86 | Initializing global parameters
INFO flwr 2025-05-29 18:05:08,232 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2025-05-29 18:05:16,942 | server.py:277 | Received initial parameters from one random client
INFO flwr 2025-05-29 18:05:16,942 | server.py:88 | Evaluating initial parameters
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
2025-05-29 18:05:16.951323: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
  1/685 ━━━━━━━━━━━━━━━━━━━━ 3:59 350ms/step - accuracy: 0.0312 - loss: 1.63 29/685 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.0060 - loss: 1.6416  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.0064 - loss: 1.6410
INFO flwr 2025-05-29 18:05:19,203 | server.py:91 | initial parameters (loss, other metrics): 1.641514539718628, {'accuracy': 0.023571331053972244}
INFO flwr 2025-05-29 18:05:19,204 | server.py:101 | FL starting
DEBUG flwr 2025-05-29 18:05:21,459 | server.py:218 | fit_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:25,181 | server.py:232 | fit_round 1 received 3 results and 0 failures
WARNING flwr 2025-05-29 18:05:25,189 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:56 258ms/step - accuracy: 1.0000 - loss: 0.08 35/685 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9996 - loss: 0.0875  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9872 - loss: 0.1284
INFO flwr 2025-05-29 18:05:26,996 | server.py:119 | fit progress: (1, 0.3915799856185913, {'accuracy': 0.9043442606925964}, 7.79256820000046)
DEBUG flwr 2025-05-29 18:05:26,997 | server.py:168 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:27,194 | server.py:182 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2025-05-29 18:05:27,194 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2025-05-29 18:05:27,194 | server.py:218 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:28,799 | server.py:232 | fit_round 2 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:48 247ms/step - accuracy: 1.0000 - loss: 0.04 39/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9972 - loss: 0.0671  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9883 - loss: 0.0881
INFO flwr 2025-05-29 18:05:30,584 | server.py:119 | fit progress: (2, 0.2116309404373169, {'accuracy': 0.9427161812782288}, 11.380167999999685)
DEBUG flwr 2025-05-29 18:05:30,585 | server.py:168 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:30,786 | server.py:182 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:30,787 | server.py:218 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:32,550 | server.py:232 | fit_round 3 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:34 226ms/step - accuracy: 1.0000 - loss: 0.04 38/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9789 - loss: 0.0947  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9754 - loss: 0.1111
INFO flwr 2025-05-29 18:05:34,276 | server.py:119 | fit progress: (3, 0.20422591269016266, {'accuracy': 0.943584144115448}, 15.071718899998814)
DEBUG flwr 2025-05-29 18:05:34,276 | server.py:168 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:34,492 | server.py:182 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:34,492 | server.py:218 | fit_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:36,344 | server.py:232 | fit_round 4 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:37 230ms/step - accuracy: 1.0000 - loss: 0.05 41/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9790 - loss: 0.0906  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9759 - loss: 0.1055
INFO flwr 2025-05-29 18:05:38,282 | server.py:119 | fit progress: (4, 0.17752012610435486, {'accuracy': 0.9521264433860779}, 19.077524299998913)
DEBUG flwr 2025-05-29 18:05:38,282 | server.py:168 | evaluate_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:38,485 | server.py:182 | evaluate_round 4 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:38,486 | server.py:218 | fit_round 5: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:40,164 | server.py:232 | fit_round 5 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:38 232ms/step - accuracy: 0.9688 - loss: 0.03 36/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9802 - loss: 0.0603  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9821 - loss: 0.0718
INFO flwr 2025-05-29 18:05:41,970 | server.py:119 | fit progress: (5, 0.1513078361749649, {'accuracy': 0.9569686055183411}, 22.765201599999273)
DEBUG flwr 2025-05-29 18:05:41,970 | server.py:168 | evaluate_round 5: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:42,214 | server.py:182 | evaluate_round 5 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:42,214 | server.py:218 | fit_round 6: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:44,005 | server.py:232 | fit_round 6 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:44 241ms/step - accuracy: 1.0000 - loss: 0.05 38/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9794 - loss: 0.0764  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9779 - loss: 0.0850
INFO flwr 2025-05-29 18:05:45,837 | server.py:119 | fit progress: (6, 0.14852099120616913, {'accuracy': 0.9581106305122375}, 26.632437299998855)
DEBUG flwr 2025-05-29 18:05:45,837 | server.py:168 | evaluate_round 6: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:46,086 | server.py:182 | evaluate_round 6 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:46,087 | server.py:218 | fit_round 7: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:47,839 | server.py:232 | fit_round 7 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:37 230ms/step - accuracy: 1.0000 - loss: 0.00 37/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9891 - loss: 0.0529  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9828 - loss: 0.0695
INFO flwr 2025-05-29 18:05:49,604 | server.py:119 | fit progress: (7, 0.14450125396251678, {'accuracy': 0.959572434425354}, 30.399307100000442)
DEBUG flwr 2025-05-29 18:05:49,604 | server.py:168 | evaluate_round 7: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:49,837 | server.py:182 | evaluate_round 7 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:49,837 | server.py:218 | fit_round 8: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:51,595 | server.py:232 | fit_round 8 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:37 231ms/step - accuracy: 1.0000 - loss: 0.02 37/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9862 - loss: 0.0660  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9807 - loss: 0.0750
INFO flwr 2025-05-29 18:05:53,396 | server.py:119 | fit progress: (8, 0.13182863593101501, {'accuracy': 0.9621762633323669}, 34.19249399999899)
DEBUG flwr 2025-05-29 18:05:53,397 | server.py:168 | evaluate_round 8: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:53,630 | server.py:182 | evaluate_round 8 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:53,632 | server.py:218 | fit_round 9: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:55,513 | server.py:232 | fit_round 9 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 2:39 233ms/step - accuracy: 1.0000 - loss: 0.03 36/685 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step - accuracy: 0.9812 - loss: 0.0655  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9802 - loss: 0.0742
INFO flwr 2025-05-29 18:05:57,372 | server.py:119 | fit progress: (9, 0.1322103887796402, {'accuracy': 0.9614453315734863}, 38.16744859999926)
DEBUG flwr 2025-05-29 18:05:57,372 | server.py:168 | evaluate_round 9: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:57,602 | server.py:182 | evaluate_round 9 received 3 results and 0 failures
DEBUG flwr 2025-05-29 18:05:57,602 | server.py:218 | fit_round 10: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:05:59,411 | server.py:232 | fit_round 10 received 3 results and 0 failures
C:\Users\hperu\anaconda3\envs\flwr_tf\lib\site-packages\keras\src\layers\core\dense.py:93: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)
  1/685 ━━━━━━━━━━━━━━━━━━━━ 3:53 342ms/step - accuracy: 1.0000 - loss: 0.04 33/685 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9869 - loss: 0.0617  685/685 ━━━━━━━━━━━━━━━━━━━━ 1s 1ms/step - accuracy: 0.9833 - loss: 0.0649
INFO flwr 2025-05-29 18:06:01,380 | server.py:119 | fit progress: (10, 0.11992958188056946, {'accuracy': 0.9653282165527344}, 42.175758600000336)
DEBUG flwr 2025-05-29 18:06:01,381 | server.py:168 | evaluate_round 10: strategy sampled 3 clients (out of 3)
DEBUG flwr 2025-05-29 18:06:01,622 | server.py:182 | evaluate_round 10 received 3 results and 0 failures
INFO flwr 2025-05-29 18:06:01,622 | server.py:147 | FL finished in 42.41855360000045
INFO flwr 2025-05-29 18:06:01,622 | app.py:218 | app_fit: losses_distributed [(1, 0.39153003692626953), (2, 0.20528550446033478), (3, 0.20129388074080148), (4, 0.17170348266760507), (5, 0.1434115171432495), (6, 0.14259813725948334), (7, 0.13667243719100952), (8, 0.12609442820151648), (9, 0.12397725880146027), (10, 0.11208732922871907)]
INFO flwr 2025-05-29 18:06:01,624 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2025-05-29 18:06:01,624 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2025-05-29 18:06:01,624 | app.py:221 | app_fit: losses_centralized [(0, 1.641514539718628), (1, 0.3915799856185913), (2, 0.2116309404373169), (3, 0.20422591269016266), (4, 0.17752012610435486), (5, 0.1513078361749649), (6, 0.14852099120616913), (7, 0.14450125396251678), (8, 0.13182863593101501), (9, 0.1322103887796402), (10, 0.11992958188056946)]
INFO flwr 2025-05-29 18:06:01,624 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.023571331053972244), (1, 0.9043442606925964), (2, 0.9427161812782288), (3, 0.943584144115448), (4, 0.9521264433860779), (5, 0.9569686055183411), (6, 0.9581106305122375), (7, 0.959572434425354), (8, 0.9621762633323669), (9, 0.9614453315734863), (10, 0.9653282165527344)]}
(flwr_tf) PS C:\Users\hperu\OneDrive\Desktop\fl\FLOWER\6C_20rounds>